src/roiextractors/extractors/tiffimagingextractors/scanimagetiff_utils.py

```diff
@@ -119,18 +125,38 @@ def parse_metadata(metadata: dict) -> dict:
     active_channels = parse_matlab_vector(metadata["SI.hChannels.channelsActive"])
     channel_indices = np.array(active_channels) - 1  # Account for MATLAB indexing
     channel_names = np.array(metadata["SI.hChannels.channelName"].split("'")[1::2])
+    scan_line_rate = 1 / float(metadata["SI.hRoiManager.linePeriod"])
     channel_names = channel_names[channel_indices].tolist()
     num_channels = len(channel_names)
-    if "RoiGroups" in metadata.keys():
-        roi_metadata = metadata["RoiGroups"]
```

Comment:
Aren't there still some files that don't have "RoiGroups"?

Reply:
It was just a safety measure. But tbh, in both scanimage_20220801_single.tif and scanimage_20220923_roi.tif testing files we do have "RoiGroups", so maybe it's also safe to remove this condition

Reply:
interesting, even the 'no_roi' file Lawrence provided (scanimage_20220923_noroi.tif) has RoiGroups.  So it is probably safe to remove the condition.